include/z64save.h

```diff
@@ -603,8 +603,11 @@ typedef enum {
 #define WEEKEVENTREG_08_10 PACK_WEEKEVENTREG_FLAG(8, 0x10)
 #define WEEKEVENTREG_RECEIVED_DOGGY_RACETRACK_HEART_PIECE PACK_WEEKEVENTREG_FLAG(8, 0x20)
 
-// Related to final hours
-#define WEEKEVENTREG_08_40 PACK_WEEKEVENTREG_FLAG(8, 0x40)
+// Set at the start and end of the Clock Tower opening cutscene as well as when the Clock Tower's
+// exterior gear is spawned after midnight on the Final Day. Thus, it is possible for this to be
+// unset, even if the player is in the Final Hours, if they did not watch the cutscene and haven't
+// been to an area where the Clock Tower is spawned.
+#define WEEKEVENTREG_CLOCK_TOWER_IS_OPENED PACK_WEEKEVENTREG_FLAG(8, 0x40)
```

Comment:
Which scenes actually set this? I'm seeing a lot of scenes that have `ACTOR_OBJ_TOKEIDAI` spawned in them, does every variant of the clock tower components set this no matter which scene it's spawned in?

Reply:
Ah, a lot of those are wall clock variants. You need to have exterior gears to set this. The following scenes have those:
- Termina Field
- North Clock Town
- South Clock Town
- West Clock Town
- East Clock Town

Should I try to work these into the comment? I figured that just saying "you need to have this specific variant spawned" would be enough, but maybe it could be clarified.

Reply:
From memory, correct me if I'm wrong, but if you're in east clock town when midnight passes, the cutscene doesn't play. It's only when you're in South clock town (and maybe termina?). Is this flag still set despite not playing the cutscene and not being initialized? I could be misremembering though

Reply:
There are three relevant cases.

1. You watch the cutscene (after the first cycle, this only happens if you're standing in South Clock Town):

https://github.com/zeldaret/mm/assets/12245827/c5a89e3c-3919-4468-b0b6-e7c3eb194715

In this case, `ObjTokeidai_TowerOpening_EndCutscene` is responsible for setting the weekeventreg.

2. You don't watch the cutscene, but you can see the Clock Tower:

https://github.com/zeldaret/mm/assets/12245827/c2d230b5-77d4-49fd-866b-89c35c32ce26

In this case, `ObjTokeidai_TowerClock_Idle` is responsible for setting the weekeventreg. It does so at the same time it starts the tower opening "transformation" sequence. It needs to do this so that `ObjTokeidai_TowerOpening_Start` will make the "transformation" happen, since it will only start the opening sequence if the right actor cue is present *or* if the weekeventreg is already set.

3. You are in a place where the Clock Tower isn't visible:

https://github.com/zeldaret/mm/assets/12245827/0c5714fc-fa25-42e1-8d8b-b7e9016def70

In this case, as soon as you load a scene where the exterior gear is present (which is to say, as soon as you load Termina Field or any of the four main Clock Town areas), `ObjTokeidai_ExteriorGear_Init` will set the weekeventreg for you.

Does that answer your question? Let me know if you have additional questions too. I'm not really sure how to pack all this into a comment.

Reply:
I tried to condense this discussion into the comment; let me know what you think. Hopefully, that makes it clearer when this is and isn't set.

Reply:
Thanks for the explanation. Yeah, I think that's clearer.